### DISCUSSION TITLE
How to retrieve the python source code which was used to train and log the model?

### DISCUSSION THREAD

**Discussion by danielmarostica** [mlflow/mlflow#7995] [Q&A] [answered]
I would like to know how to access the `train.py` file which I have used to train my model. I couldn't find it in the artifacts. Is there a way to store it along with other artifacts?

**Comment by JHibbard** [comment, marked as answer]
Hi danielmarostica,

The Git commit hash used for the run, if it was run from an MLflow Project, should be automatically recorded as the [`Code Version`](https://mlflow.org/docs/latest/tracking.html#concepts). If your runs aren't using the MLflow Project functionality, you'll have to log the Git commit hash manually with something like `mlflow.log_param` or as a tag.

Hope that helps!
Jim
